[tests] Automate Jupyter notebook tests using nbmake.

ChrisCummins opened the pull request:
This uses the nbmake pytest plugin to run the Jupyter notebooks in
`make examples-test`.

Fixes #450.

alex-treebeard commented:
hey @ChrisCummins , I just released nbmake==0.10 which should work with pytest 6.1.0. Bugs/requests welcome!

ChrisCummins commented:
> hey @ChrisCummins , I just released nbmake==0.10 which should work with pytest 6.1.0. Bugs/requests welcome!

Thanks for the heads up @alex-treebeard!

Cheers,
Chris

ChrisCummins committed:
[examples] Unify metadata in Jupyter notebooks.

ChrisCummins committed:
[examples] Add "Open in Colab" button to notebooks.

ChrisCummins committed:
[examples] Fix notebook code style.

ChrisCummins committed:
[tests] Automate Jupyter notebook tests using nbmake.

This uses the nbmake pytest plugin to run the Jupyter notebooks in
`make examples-test`.

Fixes #450.

ChrisCummins committed:
[examples] Tweak requirements.txt.

* Bump ray dependency 1.8 -> 1.9 after having problems installing 1.8.

* Don't specify numpy version (#501).
